sdk/tests/tutorials/valuation/test_valuation.py

```diff
@@ -0,0 +1,226 @@
+        """
+
+        return models.ConfigurationRecipe(
+            scope=recipe_scope,
+            code=recipe_code,
+            market=models.MarketContext(
+                market_rules=[],
+                suppliers=models.MarketContextSuppliers(equity="Lusid"),
+                options=models.MarketOptions(
+                    default_supplier="Lusid",
+                    default_instrument_code_type="LusidInstrumentId",
+                    default_scope=TestDataUtilities.tutorials_scope,
+                ),
+            ),
+        )
+
+    def upsert_recipe_request(self, configuration_recipe) -> None:
+        """
+        Structures a recipe request and upserts it into LUSID
+        :param ConfigurationRecipe configuration_recipe: Recipe configuration
+        :return: None
+        """
+
+        upsert_recipe_request = models.UpsertRecipeRequest(configuration_recipe)
+        self.recipes_api.upsert_configuration_recipe(upsert_recipe_request)
+
+    def create_aggregation_request(
+        self, inline_recipe=None, recipe_scope=None, recipe_code=None
+    ) -> lusid.models.AggregationRequest:
+        """
+        Creates an aggregation request that aggregates valuation results based on
+        provided inline recipe or recipe identifiers when using an upserted one
+        :param ConfigurationRecipe inline_recipe: A configured recipe to be used inline
+        :param str recipe_scope: The scope for an already upserted recipe
+        :param str recipe_code: The code for an already upserted recipe
+        :return: AggregationRequest
+        """
+
+        if recipe_scope:
+            in_line_recipe = self.create_configuration_recipe(recipe_scope, recipe_code)
+            self.upsert_recipe_request(in_line_recipe)
+            recipe_id = models.ResourceId(scope=recipe_scope, code=recipe_code)
+        else:
+            recipe_id = None
+
+        return models.AggregationRequest(
+            recipe_id=recipe_id,
+            inline_recipe=inline_recipe,
+            metrics=[
+                models.AggregateSpec("Instrument/default/Name", "Value"),
+                models.AggregateSpec("Holding/default/PV", "Proportion"),
+                models.AggregateSpec("Holding/default/PV", "Sum"),
+            ],
+            group_by=["Instrument/default/Name"],
+            effective_at=self.effective_date,
+        )
+
+    @parameterized.expand(
+        [
+            [
+                "Test valuation with an aggregation request containing a recipe inline",
+                create_configuration_recipe("self", "TestRecipes", "SimpleQuotes"),
+                None,
+                None,
+            ],
+            [
+                "Test valuation with an aggregation request using an already upserted recipe",
+                None,
+                "TestRecipes",
+                "SimpleQuotes",
+            ],
+        ]
+    )
+    def test_aggregation(self, _, in_line_recipe, recipe_scope, recipe_code) -> None:
```

Comment:
need to add in https://github.com/finbourne/lusid-features-python too, which may need adding a new one to https://gitlab.finbourne.com/salesengineering/lusid-features/-/blob/master/features.yml

Reply:
I think there is an issue currently with @lusidfeature decorators, @slacki123 is looking into a fix. I will update it as a comment for now, which I can then amend and also update a feature code to each example + yaml file.

Reply:
Actually it works in this instance, so I've added it. Having a look at lusidfeature as well. @rizwansaeed